doc/proposals/openshift-4.3/operator-testing-tool.md

```diff
@@ -0,0 +1,240 @@
+Today, the scorecard output shows a percentage of tests that were successful to the end user. This story is to change the scorecard output to show a pass or fail for each test that is run in the output instead of a success percentage. 
+
+The exit code of scorecard would be 0 if all tests passed. The exit code would be non-zero if tests failed. With this change scores are essentially replaced with a list of pass/fail(s).
+
+The scorecard would by default run all tests regardless if a single test fails.  Using a CLI flag such as below would cause the test execution to stop on a failure:
+ * operator-sdk scorecard -l ‘testgroup=required’ --fail-fast
+
+Tasks for this story include:
+ * change scorecard by removing earnedPoints and maximumPoints from each test
+ * change scorecard output by removing totalScorePercent, partialPass, earnedPoints, maximumPoints from scorecard output
+ * change scorecard existing tests to remove partialPass, instead only returning a pass or fail
+ * change scorecard CLI to return an exit code based on the pass/fail results from the set of executed tests 
+ * (stretch goal) add support for --fast-fail CLI flag
+
+
+Rough point estimate is 8 points.
+
+
+#### Story 2 - Change Scorecard Test Selection
+Today, the scorecard lets you select which tests are run by the plugins defined in the scorecard configuration (basic, olm). This story would change how users determine which tests are executed by scorecard.
+
+This story would introduce labels for each test and then allow the scorecard CLI to filter which tests it runs based on label selectors.
+
+Tests can fall into 3 groups: required, recommended, or optional. Tests can also be categorized as static or runtime. Labels for these groups and categories would allow a test to be more precisely specified by an end user.
+
+Possible examples of specifying what tests to run are as follows: 
+ * operator-sdk scorecard -l ‘testgroup in (required,recommended,optional)’ 
+ * operator-sdk scorecard -l ‘testtype in (runtime,static)’
+ * operator-sdk scorecard -l updatecrtest=true
+ * operator-sdk scorecard -l someplugintest=true
+
+Tasks for this story include:
+ * Update the test interfaces to support arbitrary test-defined labels
+ * Add the labels to the tests
+ * Update the CLI and scorecard library to support label selectors to filter scorecard runs
+ * A stretch goal would include the ability of the CLI to perform a dry-run of tests based on label selectors to show the user which tests would be executed for a given label selector value.
+
+Rough point estimate for this story is 8 points.
+#### Story 3 - Common Validation
+The scorecard would use a common validation codebase to verify bundle contents. This story supports a single “source of truth” and avoids duplication or variance of validation logic. The scorecard user would be able to specify a bundle on the command line, that bundle would then be validated.  For example:
+ * operator-sdk scorecard --bundle ./my-operator
```

Comment:
--config I think is better ( bundle is nomenclature added in the community repo which for me  at least was not intuitive at all)

Reply:
IIRC, `--config` is already used to define the config file used by the scorecard. I think `--bundle` is good because "bundle" has a [well-defined definition](https://github.com/operator-framework/operator-registry#manifest-format) in the context of the Operator Framework and it's what we're expecting to be passed to this flag.

One caveat/question: @dmesser Are we expecting to test the entire set of bundles that make up a channel/catalog, or are we scoping the scorecard to a single bundle at a time?